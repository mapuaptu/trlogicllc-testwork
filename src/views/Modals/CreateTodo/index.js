import Button from '@/components/Button/index.vue';

export default {
  name: 'ModalCreateTodo',
  components: {
    Button,
  },
  props: {
    onTodoSave: {
      type: Function,
      default: () => ({}),
    },
  },
  data() {
    return {
      todoName: '',
      message: '',
    };
  },
  methods: {
    onCancel() {
      return this.$emit('close');
    },
    onCreate() {
      if (!this.todoName) {
        return (this.message = 'Todo name requred');
      }

      const { error } = this.onTodoSave(this.todoName);

      if (error) {
        return (this.message = error.message);
      }

      return this.$emit('close');
    },
  },
  mounted() {
    return this.$refs.input.focus();
  },
};
