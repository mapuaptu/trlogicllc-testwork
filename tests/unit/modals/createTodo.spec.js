import CreateTodo from '@/views/Modals/CreateTodo/index.vue';
import { mount } from '@vue/test-utils';

describe('Modal createNote component', () => {
  it('1. Rendered success', async () => {
    const wrapper = await mount(CreateTodo);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
