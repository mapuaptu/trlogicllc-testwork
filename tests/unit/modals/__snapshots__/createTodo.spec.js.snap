// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal createNote component 1. Rendered success 1`] = `
<div class="modal">
  <div class="modal-header">
    Create Todo
  </div>
  <div class="modal-content"><input type="text" placeholder="Enter todo name" class="input">
    <!---->
  </div>
  <div class="modal-footer">
    <div class=""><span>
      Cancel
    </span>
      <!---->
    </div>
    <div class=""><span>
      Create
    </span>
      <!---->
    </div>
  </div>
</div>
`;
